components/fms-forwarder/src/zenoh_publisher.rs

```diff
@@ -0,0 +1,115 @@
+use crate::status_publishing::StatusPublisher;
+use async_trait::async_trait;
+use clap::{Arg, ArgMatches, Command};
+use fms_proto::fms::VehicleStatus;
+use log::{debug, warn};
+use protobuf::Message;
+use std::sync::Arc;
+use zenoh::config::Config;
+use zenoh::prelude::sync::*;
+use zenoh::publication::Publisher;
+
+const KEY_EXPR: &str = "fms/vehicleStatus";
+
+pub fn add_command_line_args(command: Command) -> Command {
+    command
+        .arg(
+            Arg::new("mode")
+		.value_parser(clap::value_parser!(WhatAmI))
+                .long("mode")
+                .short('m')
+                .help("The zenoh session mode (peer by default).")
+                .required(false),
+        )
+        .arg(
+            Arg::new("connect")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())
+                .long("connect")
+                .short('e')
+                .help("Endpoints to connect to.")
+                .required(false),
+        )
+        .arg(
+            Arg::new("listen")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())
+                .long("listen")
+                .short('l')
+                .help("Endpoints to listen on.")
+                .required(false),
+        )
+        .arg(
+            Arg::new("no-multicast-scouting")
```

Comment:
I assume that *multicast scouting* is enabled by default?
If so, then FMPOV this arg should be called `disable-multicast-scouting` and not require a value but instead use a `clap::ArgAction::SetFalse` to set the property to false.

Alternatively, it should be called `use-multicast-scouting` and use a `clap::builder::BoolishValueParser` with a default value of `true` ...

Reply:
Yes, By default zenoh router replies to multicast scouting messages for being discovered by peers and clients.This option disables this feature. This configuration name is defined in zenoh config api.

Reply:
so we are talking about the first option, keeping the long-name. Then please change the `value_parser` accordingly ...